File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/EntityImpl.java
package lt.ltrp;

import lt.ltrp.object.Entity;

/**
 * @author Bebras
 *         2015.11.29.
 */
public class EntityImpl implements Entity {

    private int uuid;

    public EntityImpl(int id) {
        this.uuid = id;
    }

    public EntityImpl() {

    }

    public int getUUID() {
        return uuid;
    }

    public void setUUID(int uuid) {
        this.uuid = uuid;
    }
}

File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/object/impl/EntityImpl.kt
package lt.ltrp.`object`.impl;

import lt.ltrp.`object`.Entity;

/**
 * @author Bebras
 *         2015.11.29.
 */
abstract class EntityImpl(uuid: Int):  Entity {

    override var UUID: Int = uuid

    constructor(): this(Entity.INVALID_ID) {

    }

    override fun equals(other: Any?): Boolean {
        return other != null && other is Entity && other.UUID == UUID
    }

    override fun hashCode(): Int {
        return UUID
    }
}
